pwnlib/libcdb.py

```diff
@@ -645,10 +787,76 @@ def search_by_sha256(hex_encoded_id, unstrip=True):
         >>> hex(ELF(filename).symbols.read)
         '0xeef40'
     """
-    return search_by_hash(hex_encoded_id, 'sha256', unstrip)
+    return search_by_hash(hex_encoded_id, 'sha256', unstrip, offline, online)
+
+def _get_libc_symbols(path):
+    """
+    Given symbol file path, the content format is as follows:
+        ...
+        puts 000xxxxx
+        read 0000xxxx
+        write 0000xxxx
+        ...
+
+    Arguments:
+        path(str):
+            symbol file path.
+
+    Returns:
+        a dictionary of the form: { name : address }, or :const:`{}`
+    """
+
+    syms = {}
 
+    with open(path, "r") as fd:
+        for x in fd.read().strip().split("\n"):
+            name, addr = x.split(" ")
```

Comment:
This would throw errors in case the format is any different.

```suggestion
            name, addr = x.split()
```

Reply:
This code is implemented exactly in accordance with the [libc-database](https://github.com/niklasb/libc-database)'s symbols file format.